tests/test_internals/test_si.py

```diff
@@ -576,5 +579,74 @@ def test_group_set_custom_description(self):
                     assert nwb.electrodes["group"][i].description == "M1 description"
 
 
+class TestSpikeInterfaceRecorders(unittest.TestCase):
+    def setUp(self):
+
+        self.test_dir = tempfile.mkdtemp()
+        self.path1 = self.test_dir + "/test_electrodes1.nwb"
+
+        self.nwbfile1 = NWBFile("session_description1", "file_id1", datetime.now())
+
+        self.RX1 = generate_recording(num_channels=4, durations=[3])
+        self.metadata_list = [dict(Ecephys={i: dict(name=i, description="desc")}) for i in ["es1", "es2"]]
+
+    def tearDown(self):
+        del self.RX1
+        shutil.rmtree(self.test_dir)
+
+    def test_non_ints_as_channel_ids(self):
+
+        channel_ids = self.RX1.get_channel_ids()
+        num_channels = self.RX1.get_num_channels()
+        non_int_channel_ids1 = ["a", "b", "c", "d"]
+        non_int_channel_ids2 = ["c", "d", "e", "f"]
+
+        self.RX_non_int_channels1 = self.RX1.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids1
+        )
+        self.RX_non_int_channels2 = self.RX1.channel_slice(
+            channel_ids=channel_ids, renamed_channel_ids=non_int_channel_ids2
+        )
+
+        values_before_rewrite = ["value_before_rewrite" for _ in range(num_channels)]
+        self.RX_non_int_channels1.set_property("property", values=values_before_rewrite)
+
+        values_after_rewrite = ["value_after_rewrite" for _ in range(num_channels)]
+        self.RX_non_int_channels2.set_property("property", values=values_after_rewrite)
+
+        # Add a property only for two
+        property_value = ["second_recoder_property" for _ in range(num_channels)]
+        self.RX_non_int_channels2.set_property("property2", values=property_value)
+
+        # Write
+        write_recording(
+            recording=self.RX_non_int_channels1, nwbfile=self.nwbfile1, metadata=self.metadata_list[0], es_key="es1"
+        )
+        # Re-write
+        write_recording(
+            recording=self.RX_non_int_channels2, nwbfile=self.nwbfile1, metadata=self.metadata_list[1], es_key="es2"
+        )
+
+        with NWBHDF5IO(str(self.path1), "w") as io:
+            io.write(self.nwbfile1)
+        with NWBHDF5IO(str(self.path1), "r") as io:
+            nwb = io.read()
+
```

Comment:
Couple minor comments about the tests...

(a) I love the use of `TestCase`, it will eventually make this file much easier to look at.

(b) Another habit we really should get into here is using the simplest types and functions possible for evaluating the functionality of targetted changes to specific functions. That is, many of these tests could be designed to operate using only `add_electrodes` and make their assertions directly on the electrode of the `nwbfile` object in memory (no need for explicit `NWBHDF5IO`.

Reply:
I agree with the spirit but in this case we are not only testing `add_electrodes` but also `add_electrical_series` were some changes also had to be made.

 I think it is better to keep the test this general test until we have more specific tests for the other parts of  `writing_recording` as future changes in other part of `writing_recording` might try to rely or assume that `nwb_file.electrodes.id` is equivalent to `channel_ids` and this test will fail alerting them that they should not. Now that said, I would agree with you that the solution for that is to write a specific test for all the different parts of `write_recording` so we can decouple testing but right now I don't think that I have enough knowledge to do that. What do you think?

Reply:
With respect to using `NWBHDF5IO`, that is, testing in memory vs read from disk, I think it makes sense. Thinking about this however I am wondering, why do we this in other tests? which kind of operations would grant a test with writing and reading to/from disk? a quick glance at the other test concerning the electrode table do not reveal evident pattern to me.

Reply:
>which kind of operations would grant a test with writing and reading to/from disk?

Tests involving data attributes; one of the biggest changes whenever you do that is the file on disk has everything written to `h5py.Dataset` objects so that's when compression/chunking actually becomes callable if you had previously specified the `data` as a list or `array`.